### ISSUE TITLE
SLURM daemons fail to start automatically, ganglia system doesn't work

### ISSUE THREAD

**Issue by GlastonburyC** [gc3-uzh-ch/elasticluster#294]
With the following configuration SLURM daemons are all down and require restart after building cluster successfully:

`pdsh -g slurm_workers sudo service slurm-llnl restart`

```
[cglastonbury@rna-seq-map-ref-bastion ~]$ cat .elasticluster/config 
[cloud/openstack]
provider=openstack
auth_url=removed
username=removed
password=removed
project_name=rna-seq-map-ref
request_floating_ip=false

[login/ubuntu]
image_user=ubuntu
image_user_sudo=root
image_sudo=True
user_key_name=craig_new
user_key_private=~/.ssh/id_rsa
user_key_public=~/.ssh/id_rsa.pub

[setup/ansible_slurm]
provider=ansible
frontend_groups=slurm_master,ganglia_master
compute_groups=slurm_workers,ganglia_monitor
ansible_ssh_pipelining=no

[setup/ansible-gridengine]
provider=ansible
frontend_groups=gridengine_master
compute_groups=gridengine_clients

[setup/ansible-pbs]
provider=ansible
frontend_groups=pbs_master,maui_master
compute_groups=pbs_clients

[cluster/slurm]
cloud=openstack
network_ids=7d326b6a-fae5-4c4f-87e4-e7b5a3628fc1
login=ubuntu
security_group=icmp-ssh-rna-seq-map-ref
image_id=1b1e4418-b710-4fd0-b143-517ac23d22a9
flavor=OrangeFS
frontend_nodes=1
compute_nodes=38

setup_provider=ansible_slurm
ssh_to=frontend
```

ganglia page not present on frontend when browsed to: > http://10.2.223.98/ganglia/

**Comment by GlastonburyC** [comment]
Eventually ganglia came up on its own.

**Comment by riccardomurri** [comment]
The critical part seems to be this one::

    PLAY [Restart SLURMd after all config is done] *********************************

    TASK [setup] *******************************************************************
    ok: [worker001]
    ok: [worker002]

    TASK [service] *****************************************************************
    task path: /home/rmurri/w/elasticluster/elasticluster/share/playbooks/roles/slurm.yml:35
    skipping: [worker001] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
    skipping: [worker002] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

    TASK [service] *****************************************************************
    task path: /home/rmurri/w/elasticluster/elasticluster/share/playbooks/roles/slurm.yml:39
    skipping: [worker001] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
    skipping: [worker002] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

    TASK [service] *****************************************************************
    task path: /home/rmurri/w/elasticluster/elasticluster/share/playbooks/roles/slurm.yml:43
    skipping: [worker001] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}
    skipping: [worker002] => {"changed": false, "skip_reason": "Conditional check failed", "skipped": true}

*One and one only* of the three `service` tasks should be executed.
However, it looks like Ansible is evaluating all conditional
expressions to "false", who knows why ...

Might be related to some changes effected in Ansible 2.1 as I didn't run into this issue with Ansible 2.0

**Comment by riccardomurri** [comment]
Definitely an issue with Ansible, as *both* these conditionals fail:

    tasks:
      - name: Check conditional #1
        debug: msg="is_ubuntu_15_10_or_later"
        when: '{{is_ubuntu_15_10_or_later}}'
      - name: Check conditional #2
        debug: msg="not is_ubuntu_15_10_or_later"
        when: 'not {{is_ubuntu_15_10_or_later}}'

Although the problem is independent of Ansible version: 2.0 and 2.1
fail in the same way.

**Comment by riccardomurri** [comment]
Hmmm.... actually, parentheses are needed after the `not` operator; with parentheses, everything seems to work as expected.  I wonder how it is possible that it has ever worked before, w/out parentheses.

**Comment by riccardomurri** [comment]
This should be fixed in commit 9833430ef5bef77f37dcac775a3c55c752922821 (later "master") -- can you please try?

**Comment by riccardomurri** [comment]
There has not been any updates on this issue since 2 months, so I'm assuming it's fixed and closing it.  Feel free to re-open if this is not the case.
